webapp/src/components/form/FormMultiFile.js

```diff
@@ -0,0 +1,130 @@
+
+
+    // add file
+    addFile = () => {
+        // variables
+        let condition = true;
+        let addError = false;
+        let handleList = this.state.fileList;
+        // check against empty fields
+        handleList.map(val => {
+            if (!val.file) {
+                condition = false
+                addError = true
+            }
+        })
+        // add item if there is no empty fields
+        if (condition) {
+            handleList.push({ name: null, file: null, delete: null, filePath: null })
+        }
+
+        this.setState({
+            addError: addError,
+            fileList: handleList
+        }, () => {
+            condition = true;
+            addError = false
+        })
+    }
+
+
+    //handle upload
+    handleUpload = (file, name, del, filePath) => {
+        // function variables
+        let handleDuplicates = false;
+        let handleList = this.state.fileList;
+
+        if (del) {
+            if (handleList.length > 1) {
+                // del file
+                let filteredList = handleList.filter((val) => {
+                    return file != val.file
+                })
+                handleList = filteredList
+            }
+            else {
+                handleList = []
+            }
+
+        }
+
+        else {
+            // Add new value
+            handleList.map(val => {
+                if (!val.file) {
+                    val.name = name;
+                    val.file = file;
+                    val.delete = del;
+                    val.filePath = filePath;
+                }
+                // test for and handle updated values
+                else if (val.file == file) {
+                    val.name = name;
+                    val.delete = del;
+                    val.filePath = filePath;
+                    handleDuplicates = true
+                }
+            })
+        }
+
+
+        // setState and Callback functions
+        this.setState({
+            fileList: handleList,
+        },  // reset function variables
+            () => {
+                if (!handleDuplicates) {
+                    this.props.uploadFile(file, handleList)
+                }
+                handleDuplicates = false;
+            })
+    }
+
+
+
+    render() {
+
+        return (
+            <div>
+                <FormGroup>
+                    {this.state.fileList.map(val => {
+                        return <MultiFileComponent className="multi-file-component"
+                            handleUpload={(file, name, del, filePath) => this.handleUpload(file, name, del, filePath)}
+                            errorText={this.props.errorText}
+                            addError={this.state.addError}
+                            del={(file) => this.del(file)}
+                            value={val}
+                        />
+                    })}
+
+                    <button className="add-file-btn" onClick={(e) => this.addFile(e)}>Add File</button>
```

Comment:
Just a note that we've recently (in the last 2/3 weeks) added translation support to the app, so now all hard-coded front-end text needs to wrapped to ensure the correct translation gets loaded. To do this, you can change this to: `this.props.t("Add File")`, then:
- Add this import: `import { withTranslation } from 'react-i18next';`
- Change the `export default FormMultiFile` at the bottom to `export default withTranslation()(FormMultiFile)
- Add the text to webapp/public/locales/en/translation.json and webapp/public/locales/fr/translation.json (you can use Google Translate to get the French text

Please do this for all hard-coded text in the UI